src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/SliderAccessibility.m

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#import "SliderAccessibility.h"
+
+#define INCREMENT 0
+#define DECREMENT 1
+
+/*
+ * Implementation of the accessibility peer for the slider role
+ */
+@implementation SliderAccessibility
+- (nullable NSString *)accessibilityLabel
+{
+    return [self accessibilityTitleAttribute];
+}
+
+- (nullable id)accessibilityValue
+{
+    return [self accessibilityValueAttribute];
+}
+
+- (BOOL)accessibilityPerformIncrement
+{
+    return [self performAccessibleAction:INCREMENT];
+}
+
+- (BOOL)accessibilityPerformDecrement
+{
+    return [self performAccessibleAction:DECREMENT];
```

Comment:
This method should upcall the java part of the patch, and may throw an exception if we will skip it.

I think we should include the java part but make it private, then re-implement this next code path to call it:
accessibilityPerformDecrement->performAccessibleAction:->CAccessibility#doAccessibleAction()->JSlider

Reply:
You're right, that's a good catch. We'll need to rework that. performAccessibleAction, however, will expect an AccessibleAction, so we need to do this differently, if we can't implement this interface in JSlider.

Reply:
CAccessibility#doAccessibleAction() can be reworked to accept Object, and then we can switch on type of the parameter, the private methods can be accessed via SwingAccessor